data-processing/tests/test_extract_definitions.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+from entities.definitions.commands.detect_definitions import *
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_model():
+    """
+    Only set up the model once, as it requires the time-consuming initialization
+    """
+    global model  # pylint: disable=global-statement
+    model = DefinitionDetectionModel()
+
+def test_term_definition_pairs(text, gold):
+    featurized_text = model.featurize(text)
+    features = [featurized_text]
+    intents, slots, slots_conf = model.predict_batch(
+        cast(List[Dict[Any, Any]], features)
+    )
+    term_definition_pairs = get_term_definition_pairs(
+            text,
+            features[0],
+            slots[0],
+            slots_conf[0]
+    )
+    assert len(term_definition_pairs) == len(gold), "Incorrect number of Term-Definitions"
+    for prediction_pair, gold_pair in zip(term_definition_pairs, gold):
+        assert text[prediction_pair.term_start:prediction_pair.term_end] == gold_pair[0], "Term Incorrect"
+        assert text[prediction_pair.definition_start:prediction_pair.definition_end] == gold_pair[1], "Definition Incorrect"
+
+def test_symbol_nickname_pairs(text, tex, gold):
+    featurized_text = model.featurize(text)
+    features = [featurized_text]
+    symbol_texs = get_symbol_texs(
+        text, tex
+    )
+    symbol_nickname_pairs = get_symbol_nickname_pairs(text, features[0], symbol_texs)
```

Comment:
If possible, let's factor the model out of this test by creating the `features` array manually as we expect to be output by the model. This has two advantages:

1. This lets us run the nickname detectors without needing to set up the model, which will let us run the tests more quickly as we improve the detection rules
2. It decouples these tests from the model, in case we start using another model that assigns different parts-of-speech, etc.

Reply:
I think it is better to not hand-code the features in tests as : 
1. The test cases (7, now) take a total of 12 secs, which is okay I think
2. Future models can add new features etc, but as we are still comparing against input (sentence) and output (term-definition pairs), these tests should still be passed.

Reply:
12 seconds is a bit long, as ideally all tests should be able to run in just a few seconds (though we're also addressing this with test tags).

Reply:
For your point #2 I feel like there should be some way to test the core logic (which is basically a set of imperative if-then rules) without requiring the loading of statistical models. Is there some other way the tests or definition detection code could be structured to permit this?